libevmasm/Assembly.cpp

```diff
@@ -1441,13 +1441,17 @@ LinkerObject const& Assembly::assembleEOF() const
 			case EOFCreate:
 			{
 				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::EOFCREATE));
-				ret.bytecode.push_back(static_cast<uint8_t>(item.data()));
+				auto const containerID = static_cast<ContainerID>(item.data());
```

Comment:
Please assert that `data()` is actually in the right range and we're not truncating it. Same for `ReturnContract`.